nixos/modules/services/web-servers/caddy/default.nix

```diff
@@ -45,21 +45,35 @@ let
       settingsFormat.generate "caddy.json" cfg.settings
     else
       let
-        Caddyfile = pkgs.writeTextDir "Caddyfile" ''
+        writeCaddyfile =
+          text:
+          pkgs.runCommandLocal "Caddyfile"
+            {
+              inherit text;
+              passAsFile = [ "text" ];
+            }
+            ''
+              mkdir -p $out
+              cp "$textPath" $out/Caddyfile
+
+              # 'validate' cannot be used for validation, due to log location access
+              # See https://github.com/caddyserver/caddy/issues/6788
+              ${lib.getExe cfg.package} adapt --config $out/Caddyfile
```

Comment:
This breaks cross-compiled nixos configs, since you can't run a non native binary on a different kind of build machine.

Reply:
Would it be enough to pass `cfg.package` as a native build input on line 53 and then just call `caddy adapt`?